src/main/kotlin/com/openlattice/BackgroundExternalDatabaseSyncingService.kt

```diff
@@ -142,7 +142,26 @@ class BackgroundExternalDatabaseSyncingService(
             }
 
         }
-        
+
+        //check if tables have been deleted in the database
+        val missingTableIds = organizationExternalDatabaseTables.keys - currentTableIds
```

Comment:
This call (and the one on line 154) are probably fine for now, but this is going to read out the entire keySet of an IMap, which can be pretty heavy and doesn't scale well. I know we do this in a few places / this is the same as the code that was here before, but as a general note, I think we should switch to a pattern of writing/executing a KeyExistsEntryProcessor for doing these checks instead of a set difference on the keySet

Reply:
https://jira.openlattice.com/browse/LATTICE-2532